core/iwasm/libraries/debug-engine/gdbserver.c

```diff
@@ -125,51 +138,157 @@ handler_packet(WASMGDBServer *server, char request, char *payload)
         packet_handler_table[(int)request].handler(server, payload);
 }
 
+static void
+process_packet(WASMGDBServer *server)
+{
+    uint8 *inbuf = (uint8 *)server->receive_ctx->receive_buffer;
+    char request;
+    char *payload = NULL;
+
+    request = inbuf[0];
+    payload = (char *)&inbuf[1];
```

Comment:
Will the inbuf's length be no smaller than 2? Shall we check the length here?

Reply:
Add a check for request. If it's '\0', that means this is an empty request, and we ignore it